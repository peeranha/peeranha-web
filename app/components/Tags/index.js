import React from 'react';
import PropTypes from 'prop-types';

const Tags = ({ chosenTags, communities, communityId }) => {
  const community = communities.filter(x => communityId === x.id)[0];
  const questionTags =
    community && community.tags.filter(x => chosenTags.includes(x.id));

  return (
    <div>
      {questionTags.map(x => (
        <span className="badge badge-secondary mr-1" key={x.name}>
          {x.name}
        </span>
      ))}
    </div>
  );
};

Tags.propTypes = {
  chosenTags: PropTypes.array,
  communities: PropTypes.array,
  communityId: PropTypes.number,
};

export default Tags;
