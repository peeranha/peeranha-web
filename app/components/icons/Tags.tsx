import React from 'react';
import IconComponent, { IconProps } from './IconComponent';

const Tags: React.FC<IconProps> = (props): JSX.Element => (
  <IconComponent
    {...props}
    id="tags"
    fill="currentColor"
    viewBox="0 0 24 24"
    size={props.size || [24, 24]}
  >
    <path
      d="m5.001 8.007 5.986 5.996.997-1-5.986-5.995-.997.999Zm11.949 3.768-5.195 5.204c-1.075 1.077-2.448 1.077-3.525 0l-6.238-6.25c-.52-.52-.997-1.228-.993-1.965l.032-6.075c.004-.824.83-1.652 1.653-1.656L8.75 1c.735-.004 1.441.473 1.962.994l6.238 6.249c1.076 1.078 1.076 2.453 0 3.53ZM18.095 7.9 11.1.905A3.089 3.089 0 0 0 8.899 0L1.89.036c-.116 0-.232.015-.345.043C.75.272.04 1.083.036 1.889L0 8.899a3.089 3.089 0 0 0 .905 2.2L7.9 18.096a3.09 3.09 0 0 0 2.185.905 3.084 3.084 0 0 0 2.185-.905l5.825-5.825c.25-.25.434-.536.581-.837.208-.426.324-.884.324-1.348a3.08 3.08 0 0 0-.905-2.184ZM6.996 6.008l5.986 5.997.998-1-5.986-5.996-.998.999ZM5.559 3.442a1.478 1.478 0 0 0-2.108 0 1.516 1.516 0 0 0 0 2.129 1.48 1.48 0 0 0 2.108 0 1.516 1.516 0 0 0 0-2.13Z"
      fill={props.fill || '#282828'}
      transform="translate(3 2)"
    />
  </IconComponent>
);

export default Tags;
