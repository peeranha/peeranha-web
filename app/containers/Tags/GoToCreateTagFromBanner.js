import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import commonMessages from 'common-messages';

import P from 'components/P';
import BaseRounded from 'components/Base/BaseRounded';
import LargeButton from 'components/Button/Contained/InfoLarge';

import communitySuggestBanner from 'images/communitySuggest.svg?inline';

import messages from './messages';

export const GoToCreateTagFromBanner = /* istanbul ignore next */ ({
  openTagForm,
}) => (
  <BaseRounded className="d-flex align-items-center py-5 my-3">
    <img
      className="mx-4"
      src={communitySuggestBanner}
      alt="communitySuggestBanner"
    />
    <div className="ml-5 mb-2">
      <P className="mb-1" fontSize="24" bold>
        <FormattedMessage {...messages.didntFindSuitableTag} />
      </P>

      <P>
        <FormattedMessage {...messages.suggestTagWhichWillbeUseful} />
      </P>

      <LargeButton className="my-4" onClick={openTagForm}>
        <FormattedMessage {...commonMessages.suggestTag} />
      </LargeButton>
    </div>
  </BaseRounded>
);

GoToCreateTagFromBanner.propTypes = {
  openTagForm: PropTypes.func,
};

export default React.memo(GoToCreateTagFromBanner);
