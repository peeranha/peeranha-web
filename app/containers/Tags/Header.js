import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import * as routes from 'routes-config';
import { TEXT_SECONDARY } from 'style-constants';

import commonMessages from 'common-messages';
import icoTagIcon from 'images/icoTag.svg?inline';
import arrowLeft from 'images/arrowLeft.svg?inline';

import communitiesHeaderFilter from 'images/communitiesHeaderFilter.svg?inline';

import H3 from 'components/H3';
import Dropdown from 'components/Dropdown';
import ArrowIcon from 'components/Dropdown/ArrowIcon';
import Span from 'components/Span';
import Ul from 'components/Ul';
import CheckedItem from 'components/Li/CheckedItem';
import { MediumImageStyled } from 'components/Img/MediumImage';

import Base from 'components/Base';
import NavigationButton from 'components/Button/Contained/Navigation';
import A from 'components/A';

import messages from './messages';
import options from './options';

const tagsRoute = routes.tags();

const H3Styled = H3.extend`
  ${ArrowIcon} {
    transform: scale(0.6);
  }
`;

const Button = /* istanbul ignore next */ ({ sorting }) => (
  <Span className="d-inline-flex align-items-center mr-2 text-capitalize" bold>
    <img
      className="mr-2"
      src={communitiesHeaderFilter}
      alt="communitiesHeaderFilter"
    />
    <FormattedMessage {...options[sorting].message} />
  </Span>
);

const Menu = /* istanbul ignore next */ ({ sortTags, sorting }) => (
  <Ul>
    {Object.keys(options).map(x => (
      <CheckedItem
        key={x}
        data-key={x}
        onClick={sortTags}
        isActive={x === sorting}
      >
        <FormattedMessage {...options[x].message} />
      </CheckedItem>
    ))}
  </Ul>
);

export const Header = /* istanbul ignore next */ ({
  goToCreateTagScreen,
  sortTags,
  sorting,
  currentCommunity,
  tagsNumber,
}) => {
  const path = window.location.pathname + window.location.hash;

  const communityTagsRoute = routes.communityTags(currentCommunity.id);
  const suggestedTagsRoute = routes.suggestedTags(currentCommunity.id);

  return (
    <div>
      <Base
        className="d-flex align-items-center justify-content-between"
        position="top"
      >
        <div>
          <A to={tagsRoute} href={tagsRoute}>
            <NavigationButton className="pl-0" isLink>
              <img className="mr-2" src={arrowLeft} alt="x" />
              <FormattedMessage {...messages.backToList} />
            </NavigationButton>
          </A>

          <A
            to={communityTagsRoute}
            href={communityTagsRoute}
            disabled={path === communityTagsRoute}
          >
            <NavigationButton isLink={path !== communityTagsRoute}>
              <FormattedMessage {...commonMessages.tags} />
            </NavigationButton>
          </A>

          <A
            to={suggestedTagsRoute}
            href={suggestedTagsRoute}
            disabled={path === suggestedTagsRoute}
          >
            <NavigationButton isLink={path !== suggestedTagsRoute}>
              <FormattedMessage {...commonMessages.voting} />
            </NavigationButton>
          </A>
        </div>

        <div>
          <NavigationButton
            onClick={goToCreateTagScreen}
            className="d-inline-flex align-items-center px-0 py-1"
            isLink
          >
            <img className="mr-1" src={icoTagIcon} alt="icoTagIcon" />
            <FormattedMessage {...commonMessages.suggestTag} />
          </NavigationButton>
        </div>
      </Base>

      <Base position="bottom">
        <H3Styled className="d-flex align-items-center justify-content-between">
          <div className="d-flex align-items-center">
            <MediumImageStyled
              src={currentCommunity.avatar}
              alt="communitiesHeader"
            />

            {currentCommunity.name}

            {tagsNumber && (
              <Span
                className="text-lowercase ml-3"
                color={TEXT_SECONDARY}
                fontSize="30"
                bold
              >
                <span>{`${tagsNumber} `}</span>
                <FormattedMessage {...commonMessages.tags} />
              </Span>
            )}
          </div>

          <Dropdown
            button={<Button sorting={sorting} />}
            menu={<Menu sortTags={sortTags} sorting={sorting} />}
            id="tags-dropdown"
            isArrowed
          />
        </H3Styled>
      </Base>
    </div>
  );
};

Button.propTypes = {
  sorting: PropTypes.string,
};

Menu.propTypes = {
  sorting: PropTypes.string,
  sortTags: PropTypes.func,
};

Header.propTypes = {
  currentCommunity: PropTypes.object,
  goToCreateTagScreen: PropTypes.func,
  sortTags: PropTypes.func,
  sorting: PropTypes.string,
  tagsNumber: PropTypes.number,
};

export default React.memo(Header);
