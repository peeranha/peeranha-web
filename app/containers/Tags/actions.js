/*
 *
 * SuggestedTags actions
 *
 */

import {
  GET_SUGGESTED_TAGS,
  GET_SUGGESTED_TAGS_SUCCESS,
  GET_SUGGESTED_TAGS_ERROR,
  GET_EXISTING_TAGS,
  GET_EXISTING_TAGS_SUCCESS,
  GET_EXISTING_TAGS_ERROR,
} from './constants';

export function getSuggestedTags({ communityId, loadMore, sorting }) {
  return {
    type: GET_SUGGESTED_TAGS,
    communityId,
    sorting,
    loadMore,
  };
}

export function getSuggestedTagsSuccess(suggestedTags, loadMore) {
  return {
    type: GET_SUGGESTED_TAGS_SUCCESS,
    suggestedTags,
    loadMore,
  };
}

export function getSuggestedTagsErr(getSuggestedTagsError) {
  return {
    type: GET_SUGGESTED_TAGS_ERROR,
    getSuggestedTagsError,
  };
}

export function getExistingTags({ communityId, loadMore, sorting, text }) {
  return {
    type: GET_EXISTING_TAGS,
    communityId,
    sorting,
    loadMore,
    text,
  };
}

export function getExistingTagsSuccess(existingTags, loadMore) {
  return {
    type: GET_EXISTING_TAGS_SUCCESS,
    existingTags,
    loadMore,
  };
}

export function getExistingTagsErr(getExistingTagsError) {
  return {
    type: GET_EXISTING_TAGS_ERROR,
    getExistingTagsError,
  };
}
