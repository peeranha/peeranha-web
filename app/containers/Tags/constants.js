export const GO_TO_CREATE_TAG_SCREEN_BUTTON_ID =
  'app_containers_GO_TO_CREATE_TAG_SCREEN_BUTTON_ID';
