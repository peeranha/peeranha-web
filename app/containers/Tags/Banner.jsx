import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import commonMessages from 'common-messages';

import Wrapper from 'components/Banner';
import Button from 'components/Button/Contained/InfoLarge';

import communitySuggestBanner from 'images/communitySuggest.svg?inline';

import messages from './messages';

import { GO_TO_CREATE_TAG_SCREEN_BUTTON_ID } from './constants';

export const Banner = ({ openTagForm }) => (
  <Wrapper>
    <img src={communitySuggestBanner} alt="tags-banner" />
    <div>
      <p>
        <FormattedMessage {...messages.didntFindSuitableTag} />
      </p>

      <p>
        <FormattedMessage {...messages.suggestTagWhichWillbeUseful} />
      </p>

      <Button
        id={`${GO_TO_CREATE_TAG_SCREEN_BUTTON_ID}_banner`}
        onClick={openTagForm}
      >
        <FormattedMessage {...commonMessages.suggestTag} />
      </Button>
    </div>
  </Wrapper>
);

Banner.propTypes = {
  openTagForm: PropTypes.func,
};

export default React.memo(Banner);
