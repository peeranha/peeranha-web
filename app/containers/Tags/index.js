/**
 *
 * Tags
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose, bindActionCreators } from 'redux';

import { DAEMON } from 'utils/constants';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import { makeSelectLocale } from 'containers/LanguageProvider/selectors';
import { makeSelectProfileInfo } from 'containers/AccountProvider/selectors';
import { selectCommunities } from 'containers/DataCacheProvider/selectors';

import { redirectToCreateTag } from 'containers/CreateTag/actions';

import LoadingIndicator from 'components/LoadingIndicator/WidthCentered';
import AsideBox from 'components/Base/Aside';

import reducer from './reducer';
import saga from './saga';
import { getSuggestedTags, getExistingTags } from './actions';
import * as selectors from './selectors';

import Header from './Header';
import Banner from './Banner';

/* eslint consistent-return: 0 */
/* eslint-disable react/prefer-stateless-function */
export class Tags extends React.Component {
  componentDidMount() {
    const {
      communityId,
      getSuggestedTagsDispatch,
      getExistingTagsDispatch,
    } = this.props;

    getExistingTagsDispatch({ communityId });
    getSuggestedTagsDispatch({ communityId });
  }

  componentDidUpdate(prevProps) {
    const { communityId, getExistingTagsDispatch, communities } = this.props;

    if (!prevProps.communities.length && communities.length) {
      getExistingTagsDispatch({ communityId });
    }
  }

  render() /* istanbul ignore next */ {
    const {
      sorting,
      currentCommunity,
      tagsNumber,
      sortTags,
      Content,
      Aside,
      redirectToCreateTagDispatch,
    } = this.props;

    if (!currentCommunity.tags.length) return <LoadingIndicator />;

    return (
      <div className="d-flex justify-content-center">
        <div className="flex-grow-1">
          <Header
            goToCreateTagScreen={redirectToCreateTagDispatch}
            sortTags={sortTags}
            sorting={sorting}
            currentCommunity={currentCommunity}
            tagsNumber={tagsNumber}
          />

          <div className="mb-3">{Content}</div>

          <Banner openTagForm={redirectToCreateTagDispatch} />
        </div>

        <AsideBox className="d-none d-xl-block">{Aside}</AsideBox>
      </div>
    );
  }
}

Tags.propTypes = {
  sorting: PropTypes.string,
  redirectToCreateTagDispatch: PropTypes.func,
  getSuggestedTagsDispatch: PropTypes.func,
  getExistingTagsDispatch: PropTypes.func,
  Aside: PropTypes.any,
  Content: PropTypes.any,
  sortTags: PropTypes.func,
  tagsNumber: PropTypes.number,
  currentCommunity: PropTypes.object,
  communities: PropTypes.array,
  communityId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

const mapStateToProps = createStructuredSelector({
  locale: makeSelectLocale(),
  profile: makeSelectProfileInfo(),
  sorting: selectors.selectSorting(),
  existingTagsLoading: selectors.selectExistingTagsLoading(),
  suggestedTagsLoading: selectors.selectSuggestedTagsLoading(),
  communities: selectCommunities(),
});

function mapDispatchToProps(dispatch) /* istanbul ignore next */ {
  return {
    getSuggestedTagsDispatch: bindActionCreators(getSuggestedTags, dispatch),
    getExistingTagsDispatch: bindActionCreators(getExistingTags, dispatch),
    redirectToCreateTagDispatch: bindActionCreators(
      redirectToCreateTag,
      dispatch,
    ),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'tags', reducer });
const withSaga = injectSaga({ key: 'tags', saga, mode: DAEMON });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(Tags);
