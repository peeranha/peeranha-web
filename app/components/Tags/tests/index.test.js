import Tags from '../index';

describe('Tags', () => {
  const props = {
    communities: [
      {
        id: 1,
        name: 'name',
        tags: [
          {
            id: 1,
            name: 'tag1',
          },
          {
            id: 2,
            name: 'tag2',
          },
          {
            id: 3,
            name: 'tag3',
          },
        ],
      },
    ],
    communityId: 1,
    chosenTags: [1, 2],
  };

  it('test, chosenTags - [... , .... , ...]', () => {
    expect(Tags(props)).toMatchSnapshot();
  });
});
