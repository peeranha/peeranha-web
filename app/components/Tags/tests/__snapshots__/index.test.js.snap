// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags test, chosenTags - [... , .... , ...] 1`] = `
<div>
  <span
    className="badge badge-secondary mr-1"
  >
    tag1
  </span>
  <span
    className="badge badge-secondary mr-1"
  >
    tag2
  </span>
</div>
`;

exports[`Tags test, communityId - null 1`] = `<div />`;
